accessible/fieldview.component.js

```diff
@@ -0,0 +1,123 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Angular2 Component that details how a Blockly.Field is
+ * rendered in the toolbox in AccessibleBlockly. Also handles any interactions
+ * with the field.
+ * @author madeeha@google.com (Madeeha Ghori)
+ */
+var blocklyApp = blocklyApp || {};
+
+blocklyApp.FieldView = ng.core
+  .Component({
+    selector: 'field-view',
+    template: `
+    <li #listItem aria-selected=false role='treeitem' [attr.aria-level]='level' *ngIf='isTextInput(field)' id='{{treeService.createId(listItem)}}'>
+      <input #input id='{{treeService.createId(input)}}' [ngModel]='field.getValue()' (ngModelChange)='field.setValue($event)'>
+      {{setLabelledBy(listItem, concatStringWithSpaces('blockly-argument-input', input.id))}}
+    </li>
+    <li #listItem aria-selected=false role='treeitem' [attr.aria-level]='level' *ngIf='isDropdown(field)' id='{{treeService.createId(listItem)}}'>
+      <label #label id='{{treeService.createId(label)}}'>current argument value: {{field.getText()}}</label>
+      <ol role='group'  [attr.aria-level]='level+1'>
+        <li #option *ngFor='#optionValue of getOptions(field)' id='{{treeService.createId(option)}}' role='treeitem' aria-selected=false [attr.aria-level]='level+1'>
+          <button #optionButton id='{{treeService.createId(optionButton)}}' (click)='handleDropdownChange(field,optionValue)'>{{optionText[optionValue]}} button</button>
+        </li>
+      </ol>
+      {{setLabelledBy(listItem, concatStringWithSpaces('blockly-argument-menu', label.id))}}
+    </li>
+    <li #listItem aria-selected=false role='treeitem' id='{{treeService.createId(listItem)}}' [attr.aria-level]='level' *ngIf='isCheckbox(field)'>
+      //TODO(madeeha):CHECKBOX
+    </li>
+    <li #listItem aria-selected=false role='treeitem' id='{{treeService.createId(listItem)}}' [attr.aria-level]='level' *ngIf='isTextField(field) && notWhitespace(field)'>
+      <label #label id='{{treeService.createId(label)}}'>
+        {{field.getText()}}
+      </label>
+      {{setLabelledBy(listItem, concatStringWithSpaces('blockly-argument-text', label.id))}}
+    </li>
+    `,
+    inputs: ['field', 'level', 'index', 'parentId'],
+  })
+  .Class({
+    constructor: [blocklyApp.TreeService, function(_service) {
+      this.optionText = {
+        keys: []
+      };
+      this.text = 'Nothing';
+      this.treeService = _service;
+    }],
+    setLabelledBy: function(item,string){
```

Comment:
Also, 'string' is a very non-descriptive argument name. What does this string represent? Call it that instead.